process_email.py

```diff
@@ -622,35 +620,50 @@ def _handle_attachment(self, part, file_name, file_path):
                 del attach_meta_info['content']
                 curr_attach['should_ignore'] = True
 
-        part_payload = part.get_payload(decode=True)
-        try:
-            if not part_payload:
-                with open(file_path, 'wb'):
-                    self._base_connector.debug_print("Payload not found, hence adding empty file")
-            else:
+        if isinstance(part.get_payload(), list):
+            try:
+                part_payload = part.get_payload()[0]
+                self._debug_print("payload content type: {0}".format(type(part_payload)))
+                with open(file_path, 'wb') as f:
+                    f.write(part_payload.as_string().encode('utf-8'))
```

Comment:
What is the expected type here? If it's an email, then per the docs https://docs.python.org/3/library/email.message.html#email.message.EmailMessage.as_string maybe this is not the best way to do this?

Reply:
The type is an email so I think you're right. In the docs it recommends to use [email.generator.Generator](https://docs.python.org/3/library/email.generator.html#email.generator.Generator) which is better for serializing messages. Will change the code to use this instead